### PR TITLE
[AIRFLOW-6140] Add missing types for some core classes.

### PR DESCRIPTION
**Pull request by potiuk**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6140
### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by potiuk** [comment]
That's the first of the following PRs after executors unentangling - adds some missing types and cleans up some of the core classes.

File: airflow/executors/executor_loader.py [hunk 1]
```diff
@@ -25,11 +25,11 @@ class ExecutorLoader:
...
-    LOCAL_EXECUTOR = "LocalExecutor"
-    SEQUENTIAL_EXECUTOR = "SequentialExecutor"
-    CELERY_EXECUTOR = "CeleryExecutor"
-    DASK_EXECUTOR = "DaskExecutor"
-    KUBERNETES_EXECUTOR = "KubernetesExecutor"
+    LOCAL_EXECUTOR: str = "LocalExecutor"
+    SEQUENTIAL_EXECUTOR: str = "SequentialExecutor"
+    CELERY_EXECUTOR: str = "CeleryExecutor"
+    DASK_EXECUTOR: str = "DaskExecutor"
+    KUBERNETES_EXECUTOR: str = "KubernetesExecutor"
```
Comment (kaxil) [inline airflow/executors/executor_loader.py]: Do we need to add `str` here? this are static variables that would always contain string ??

May be yes, may be no.

Just laying out my thought

File: airflow/models/baseoperator.py [hunk 2]
```diff
@@ -869,11 +877,16 @@ def get_task_instances(self, start_date=None, end_date=None, session=None):
...
+    def get_flat_relative_ids(self,
+                              upstream: bool = False,
+                              found_descendants: Optional[Set[str]] = None) -> Set[str]:
         """
-        Get a flat list of relatives' ids, either upstream or downstream.
+        Get a flat set of relatives' ids, either upstream or downstream.
         """
 
+        if not self._dag:
+            return set()
```
Comment (kaxil) [inline airflow/models/baseoperator.py:893]: When is this helpful?

File: airflow/executors/executor_loader.py (hunk 1 above)
Comment (potiuk) [reply airflow/executors/executor_loader.py]: Not really :)

File: airflow/models/baseoperator.py (hunk 2 above)
Comment (potiuk) [reply airflow/models/baseoperator.py:893]: It's only when the dag has never been set. We could also throw an exception here (or assert :)), but I think returning empty set is a better choice (because no Dag has no relative ids :D).

With MyPy we now detect many places where None'able (i.e. Optional) field is used to call a method which is great because it really avoids having a mysterious errors during development.

**Comment by potiuk** [comment]
Hey @kaxil - re-reviewed and corrected. Pls take a look .

File: .travis.yml [hunk 3]
```diff
@@ -24,7 +24,7 @@ env:
     - MOUNT_LOCAL_SOURCES="false"
     - AIRFLOW_MOUNT_HOST_VOLUMES_FOR_STATIC_CHECKS="true"
     - FORCE_ANSWER_TO_QUESTIONS="yes"
-    - AIRFLOW_CI_SILENT="true"
+    - AIRFLOW_CI_SILENT="false"
```
Comment (mik-laj) [inline .travis.yml]: It looks like unrelated changes.

Comment (potiuk) [reply .travis.yml]: True. It was for debugging of a problem. Will revert.

**Review by feluelle — APPROVED** [review]
LGTM 👍 (after you revert the `AIRFLOW_CI_SILENT` change)
